addons/account_edi/models/account_payment.py

```diff
@@ -7,6 +7,73 @@
 class AccountPayment(models.Model):
     _inherit = 'account.payment'
 
+    edi_show_cancel_button = fields.Boolean(
+        compute='_compute_edi_show_cancel_button')
+    edi_show_abandon_cancel_button = fields.Boolean(
+        compute='_compute_edi_show_abandon_cancel_button')
+
+    @api.depends(
+        'state',
+        'edi_document_ids.state')
+    def _compute_edi_show_cancel_button(self):
+        for payment in self:
+            if payment.state != 'posted':
+                payment.edi_show_cancel_button = False
+                continue
+
+            payment.edi_show_cancel_button = any([doc.edi_format_id._needs_web_services()
```

Comment:
could call the compute on the move_id as well...  (as it inherits, are you sure the field is not there already on the account.payment?  (could also be a related: move_id.edi_show_cancel_button)

Reply:
@MerlinGuillaume I don't see any reason to go check the reconciled_invoice_ids.  If the doc.state is sent, it is that these were there.  (unless you would need them in the api call to the PAC or something)  You can just remove L26-27.  

But you are right that in v15 we still need another logic for the button on payments. (but in v16 we should just be able to call the method on account.move)